diff --git a/bindings/java/Makefile b/new/bindings/java/Makefile
index ba9fa56..1fc0d23 100644
--- a/bindigns/java/Makefile
+++ b/bindings/java/Makefile
@@ -49,7 +49,7 @@ tests: capstone_class jna
 		TestPpc.java TestSparc.java TestSystemz.java TestM680x.java
 
 gen_const:
-	cd ../ && $(PYTHON2) const_generator.py java
+	cd ../ && python3.12 const_generator.py java
 
 jna:
 	@if [ ! $(JNA) ]; then echo "*** Unable to find JNA ***"; exit 1; fi
